src/community/mod.rs

```diff
@@ -84,26 +84,23 @@ pub struct CommunityPermissions {
 
 impl Community {
     pub fn validate(&self) {
-        if self.handle.len() < 3 || self.handle.len() > 40 {
+        if self.handle.chars().count() < 3 || self.handle.chars().count() > 40 {
```

Comment:
Your code might be optimized, but it might also not optimized and then you would have twice than necessary times computing the `.count()`, so I suggest we use `matches!` everywhere:

```suggestion
        if !matches(self.handle.chars().count(),  3..=40) {
```